server/src/future.rs

```diff
@@ -33,16 +33,17 @@ use tokio::sync::{watch, OwnedSemaphorePermit, Semaphore, TryAcquireError};
 /// Represent a stop handle which is a wrapper over a `multi-consumer receiver`
 /// and cloning [`StopHandle`] will get a separate instance of the underlying receiver.
 #[derive(Debug, Clone)]
-pub(crate) struct StopHandle(watch::Receiver<()>);
+pub struct StopHandle(watch::Receiver<()>);
 
 impl StopHandle {
```

Comment:
So here, the user has to create the tx/rx themselves, and then construct the stop handle and server handle. I wonder if there's a better way to hide the implementation details around this shutting down stuff so that the user isn't making txs/rxs etc. Or does the user benefit from being able to make these themselves?

Reply:
It would be great to get rid of that but it's tricky because once the `tx handle` is dropped then the server is stopped.
So it would be hard to hide that without running to the server forever i.e, not be able to stop it.

Perhaps we can provide a helper function that avoid constructing the channel at least.

Reply:
I like the function better yup, and it means we can change the internals without worry :)